pkg/pidfile/pidfile.go

```diff
@@ -0,0 +1,51 @@
+package pidfile
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+)
+
+type PidFile struct {
+	path string
+}
+
+func checkPidFileAlreadyExists(path string) error {
+	if pidString, err := ioutil.ReadFile(path); err == nil {
+		if pid, err := strconv.Atoi(string(pidString)); err == nil {
+			if _, err := os.Stat(filepath.Join("/proc", string(pid))); err == nil {
+				return fmt.Errorf("pid file found, ensure docker is not running or delete %s", path)
+			}
+		}
+	}
+	return nil
+}
+
+func New(path string) (file *PidFile, err error) {
+	if err = checkPidFileAlreadyExists(path); err != nil {
+		return nil, err
+	}
+
+	createdFile, err := os.Create(path)
```

Comment:
That sounds nice indeed, and it would shorten the code a lot. However, when I tried this, I ran into a conversion from `int` to `[]byte`, for which I couldn't find a really nice solution.
Do you have a suggestion for that?